### DISCUSSION TITLE
How to run .TP Program

### DISCUSSION THREAD

**Discussion by Sanchitkedia** [gavanderhoorn/dominh#22] [Q&A] [answered]
I have ".TP" programs that read register values and execute moves. How can I run those? I am able to do the same on Windows with PCDK, but I wanted to do it for Python and Linux.

**Comment by gavanderhoorn** [comment, marked as answer]
So, first the obligatory disclaimer (from the readme):

> This is a poor man's version of a subset of the RPC functionality provided by the (Windows-only) Fanuc PCDK [..]

right now, what you ask is not directly supported by Dominh.

I can think of a couple of work-arounds though:

1. setup remote start of programs, select the `.tp` you want to run, trigger a remote start
1. write a small Karel program which you start at controller startup, have it monitor some `FLAG` and on signal-up start your `.tp` program
1. do the same (ie: monitoring `FLAG`s) but with a `.tp` program

finally: *officially* you are not supposed to start any programs with motion remotely using the Web Server Enhancements -- which would allow you to directly call the necessary KCL functions, removing the need to use any of the work-arounds I describe above.

You can configure your controller to allow that though. I don't have access to the manuals right now, so I can't tell you exactly how to do it, but there should be enough information in the *FANUC Robot series - Ethernet Function - Operator's Manual*.
